pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,660 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamicresources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
```

Comment:
ok, in this package I have a say :)

I prefer we don't use these libraries and use `t.Errorf` and `t.Fatalf`. They allow for more clarity and flexibility when paired with `cmp.Diff`.

Reply:
Hmm, this is not straightforward.

```
	if diff := cmp.Diff(expected.status, status); diff != "" {
		t.Errorf("status not as expected: want %+v, got %+v, diff (-want +got):\n%s", expected.status, status, diff)
	}
```

That now fails with:
```
            dynamicresources_test.go:345: status not as expected: want &{code:1 reasons:[look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found] err:0xc00044b5e0 failedPlugin:}, got &{code:1 reasons:[look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found] err:0xc000495130 failedPlugin:}, diff (-want +got):
                  (*framework.Status)(
                - 	&⟪0xc000041440⟫{
                - 		code: 1,
                - 		reasons: []string(⟪ptr:0xc00044b620, len:1, cap:1⟫{
                - 			`look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found`,
                - 		}),
                - 		err: &⟪0xc00044b5e0⟫"errors".errorString{
                - 			s: `look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found`,
                - 		},
                - 	},
                + 	&⟪0xc0005d1080⟫{
                + 		code: 1,
                + 		reasons: []string(⟪ptr:0xc000495270, len:1, cap:1⟫{
                + 			`look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found`,
                + 		}),
                + 		err: &⟪0xc000495130⟫"errors".errorString{
                + 			s: `look up resource class: resourceclass.resource.k8s.io "my-resource-class" not found`,
                + 		},
                + 	},
                  )
```

The pointer values are different, but I thought that would be ignored by default?

Reply:
cmp uses `errors.Is`, that's why.

I guess we can leave this cleanup for another PR.

Reply:
Yes, please :sweat_smile: 

I created https://github.com/kubernetes/kubernetes/issues/113774 to track this.